text/0000-volta-use.md

```diff
@@ -0,0 +1,103 @@
+
+- If `--global` was set -> use `volta install` behavior
+- If `--pin` was set -> use `volta pin` behavior
+- If _both_ were set, error (this can be handled by Clap I believe by making the two flags exclusive)
+- If neither flag was set:
+  - If we are in a pinned project (one that already has a `"volta"` key in `package.json`) -> use `volta pin` behavior.
+  - Otherwise, use `volta install` behavior.
+
+# Critique
+
+[critique]: #critique
+
+## Confusion around omitting version
+
+One nice part about the existing `volta install` and `volta pin` commands is that they naturally work well with choosing a tool without selecting a version. For example, running `volta install node` makes intuitive sense as "pick whatever version and make it available." By contrast, `volta use node` doesn't immediately (to me, at least!) jump out as _picking_ a version of Node. `volta use node@lts` or similar, _do_ make sense, but omitting the version feels a little more awkward with `use` as the verb.
+
+## Changing only the verb for `install`
+
+One alternative would be to only change the verb for `install`, leaving `pin` completely untouched. This would be the least disruptive change, however it's very dependent on the verb / command that we choose. With the proposal of `volta use`, it would feel very unintuitive to run `volta use node@21` in a project, and then immediately get a message that it's set as the global but your local settings override. As a user, I would expect `volta use node@21` to mean the next call to `node --version` to display Node 21 is running. In order to make that work as users expect, we need to unify the two so that `volta use` _does_ actually change the Node for the current context.
+
+Another difficulty of the unified behavior is that it's more difficult to explain when users run into edge cases. Right now there's a clear separation between "local" and "global" with two different commands. Unifying them means we'll need to clearly document how `volta use` behaves in each situation, and how to override that (with the `--global` or `--pin` flags).
+
+## `volta default` or `volta global`
+
+In the vein of replacing the verb for `install`, I also considered `default` or `global` for the command. The major issue with those is that they're adjectives, rather than verbs, so it's not immediately clear what the command means. Does `volta default node` mean "choose a default node" or "list the current default node"? That confusion is ultimately why I decided against using either of these for the command.
+
+## `volta set`
+
+Another possibility for the command would be to make it more verbose, e.g. `volta set default node`. That solves the problems around `volta default` or `volta global` by adding a verb to make it very clear what you are doing. However, it makes the commands a fair bit more cumbersome to use, which I feel is ultimately a net negative compared to a single command.
```

Comment:
One idea here may be to use `volta set` as the new command, rather than `volta use`. That makes it even more clear that the important part of the operation is setting the version, however it breaks from users expectations from other tools.